### PR TITLE
Cleanup webappconfiguration

### PR DESCRIPTION
**Pull request by gregw**
cleanup webappconfiguration

### CODE REVIEW

File: examples/async-rest/async-rest-webapp/src/main/webapp/WEB-INF/jetty-web.xml [hunk 1]
```diff
@@ -9,7 +9,7 @@ org.eclipse.jetty.servlet.WebApplicationContext object
 -->
 
 <Configure class="org.eclipse.jetty.webapp.WebAppContext">
-  <Get class="org.eclipse.jetty.util.log.Log" name="rootLogger">
-    <Call name="warn"><Arg>async-rest webapp is deployed. DO NOT USE IN PRODUCTION!</Arg></Call>
+  <Get name="servletContext">
+    <Call name="log"><Arg>The async-rest webapp is deployed. DO NOT USE IN PRODUCTION!</Arg><Arg></Arg></Call>
```
Comment (joakime) [inline examples/async-rest/async-rest-webapp/src/main/webapp/WEB-INF/jetty-web.xml]: Is this a WARNING level event?
If not, then I don't like this change. :-)

Comment (gregw) [reply examples/async-rest/async-rest-webapp/src/main/webapp/WEB-INF/jetty-web.xml]: It is a warning level event and by passing the null second arg (a Throwable), it remains a warning level event.

**Comment by gregw** [comment]
Note that this PR is really a draft until we work out how to fix the OSGi tests.  @janbartel any ideas yet?

**Comment by gregw** [comment]
@janbartel nudge

File: jetty-webapp/src/main/java/org/eclipse/jetty/webapp/WebAppConfiguration.java [hunk 2]
```diff
@@ -32,14 +32,10 @@ public WebAppConfiguration()
...
+            "org.eclipse.jetty.servlet.StatisticsServlet",
             "org.eclipse.jetty.servlet.DefaultServlet", 
-            "org.eclipse.jetty.servlet.NoJspServlet",
-            "org.eclipse.jetty.continuation.");
-        expose( // TODO Evaluate why these are not protectAndExpose?
-            "org.eclipse.jetty.servlet.listener.",
-            "org.eclipse.jetty.alpn.");
+            "org.eclipse.jetty.servlet.NoJspServlet"
+        );
+        expose("org.eclipse.jetty.servlet.listener.");
```
Comment (sbordet) [inline jetty-webapp/src/main/java/org/eclipse/jetty/webapp/WebAppConfiguration.java]: All these are classes/packages are in `org.eclipse.jetty.servlet`.
Would it not make more sense to have a `Configuration` class in the `jetty-servlet` module rather than in `jetty-webapp`?

File: examples/async-rest/async-rest-webapp/src/main/webapp/WEB-INF/jetty-web.xml (hunk 1 above)
Comment (sbordet) [reply examples/async-rest/async-rest-webapp/src/main/webapp/WEB-INF/jetty-web.xml]: @gregw I fear that this passes an empty `String` as second parameter.
The way to pass `null` would be `<Arg />`, not `<Arg></Arg>` (which should pass an empty string). Can you double check?

File: tests/test-webapps/test-jetty-webapp/pom.xml [hunk 3]
```diff
@@ -141,6 +141,7 @@
       <groupId>org.eclipse.jetty</groupId>
       <artifactId>jetty-servlets</artifactId>
       <version>${project.version}</version>
+      <scope>provided</scope>
```
Comment (sbordet) [inline tests/test-webapps/test-jetty-webapp/pom.xml]: Why `provided`? `jetty-servlets` is supposed to be portable across containers (it's not, but I think that's the idea), so it should really be in `WEB-INF/lib`.

File: tests/test-webapps/test-servlet-spec/test-spec-webapp/src/main/java/com/acme/test/MultiPartTest.java [hunk 4]
```diff
@@ -116,8 +116,20 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
...
+        byte buffer[] = new byte[bufferSize];
+
+        while (true)
+        {
+            int len=in.read(buffer,0,bufferSize);
+            if (len<0 )
+                break;
+            out.write(buffer,0,len);
+        }
+    }
```
Comment (sbordet) [inline tests/test-webapps/test-servlet-spec/test-spec-webapp/src/main/java/com/acme/test/MultiPartTest.java]: What's the reason to inline `IO.copy()`?

File: tests/test-webapps/test-servlet-spec/test-spec-webapp/src/main/webapp/WEB-INF/jetty-web.xml [hunk 5]
```diff
@@ -2,7 +2,10 @@
 <!DOCTYPE Configure PUBLIC "-//Mort Bay Consulting//DTD Configure//EN" "http://www.eclipse.org/jetty/configure_9_3.dtd">
 
 <Configure class="org.eclipse.jetty.webapp.WebAppContext">
+  <Get name="servletContext">
+    <Call name="log"><Arg>The test-spec webapp is deployed. DO NOT USE IN PRODUCTION!</Arg><Arg></Arg></Call>
+  </Get>
   <Get class="org.eclipse.jetty.util.log.Log" name="rootLogger">
-    <Call name="warn"><Arg>test-spec webapp is deployed. DO NOT USE IN PRODUCTION!</Arg></Call>
+    <Call name="info"><Arg>WEB-INF/lib/jetty-util.jar logging used!</Arg></Call>
```
Comment (sbordet) [inline tests/test-webapps/test-servlet-spec/test-spec-webapp/src/main/webapp/WEB-INF/jetty-web.xml]: Why this additional log line about `jetty-util.jar`? Debug leftover?

File: jetty-webapp/src/main/java/org/eclipse/jetty/webapp/WebAppConfiguration.java (hunk 2 above)
Comment (gregw) [reply jetty-webapp/src/main/java/org/eclipse/jetty/webapp/WebAppConfiguration.java]: Not really.  servlet is usable without webapps.    But configuration really about configuring webapps - specifically by a combination of  convention and the configurations we push into it from our modules.   If we wanted to move Configuration down to servlet, then we'd have to move the WebAppClassLoader (or at least an abstraction of it) down as well... and then we start having no difference between servlet and webapp.

tl;dr; any direct users of jetty-servlet configure by using API directly and probably don't have a different classloader (maybe not even a context).

File: tests/test-webapps/test-jetty-webapp/pom.xml (hunk 3 above)
Comment (gregw) [reply tests/test-webapps/test-jetty-webapp/pom.xml]: I don't think they are portable.  More over, the jetty-servlets module exposes them on the webapp classpath.  Putting them in webapp just brings in all our io, http and util classes and exposes all the classloader problems that entails.
In fact, some of the classes (eg PushCacheFilter) only work if loaded from the container.    I don't think we are in the business of providing general utility servlets for any container.

File: tests/test-webapps/test-servlet-spec/test-spec-webapp/src/main/java/com/acme/test/MultiPartTest.java (hunk 4 above)
Comment (gregw) [reply tests/test-webapps/test-servlet-spec/test-spec-webapp/src/main/java/com/acme/test/MultiPartTest.java]: because you wouldn't let me do a release of a fixed jetty-util jar :)
Once we have a fixed 9.4 jetty-util release, we can re-instate the tests that use it and then we can remove this inline (I told you it was a pain!)
I don't want to bring in jetty-10 util just for this 1 usage.

File: tests/test-webapps/test-servlet-spec/test-spec-webapp/src/main/webapp/WEB-INF/jetty-web.xml (hunk 5 above)
Comment (gregw) [reply tests/test-webapps/test-servlet-spec/test-spec-webapp/src/main/webapp/WEB-INF/jetty-web.xml]: The first one is the general warning about production.
The second one is a usage of the WEB-INF/lib/jetty-util logging... which should be the old version.... however it is currently not there, so I need to comment out this line like the others with a TODO to put it back when there is a fixed 9.4 release
